week08_pomdp/practice_pytorch.ipynb

```diff
@@ -444,12 +466,50 @@
    "source": [
     "def to_one_hot(y, n_dims=None):\n",
     "    \"\"\" Take an integer tensor and convert it to 1-hot matrix. \"\"\"\n",
-    "    y_tensor = y.to(dtype=torch.int64).view(-1, 1)\n",
+    "    y_tensor = y.to(dtype=torch.int64).contiguous().view(-1, 1)\n",
     "    n_dims = n_dims if n_dims is not None else int(torch.max(y_tensor)) + 1\n",
     "    y_one_hot = torch.zeros(y_tensor.size()[0], n_dims).scatter_(1, y_tensor, 1)\n",
     "    return y_one_hot"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Build Reproducibility function"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import random \n",
+    "def set_seed(seed_value: int=42) -> None:\n",
+    "  os.environ['PYTHONHASHSEED'] = f\"{seed_value}\"\n",
+    "  random.seed(seed_value)\n",
+    "  np.random.seed(seed_value)\n",
+    "  torch.manual_seed(seed_value)\n",
+    "\n",
+    "  # If cuda\n",
+    "  if torch.cuda.is_available():\n",
+    "    torch.cuda.manual_seed(seed_value)\n",
+    "    torch.cuda.manual_seed_all(seed_value)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# adding lambda λ to make compromise between bias and variance.\n",
+    "lamda = 0.92 # best value according to (Schulman et al., 2018) if you are going to work with GAE\n",
+    "ENT_COEF = 0.01\n",
+    "GRADIENT_COEF = 0.5"
```

Comment:
This value is not used later.

Reply:
USED!!!

![image](https://user-images.githubusercontent.com/72295771/205723250-5f2bf64c-1215-4a7d-bd30-51b7a2583d9e.png)

![image](https://user-images.githubusercontent.com/72295771/205723290-62c7d662-4d43-4837-b9aa-2149a9df7d9a.png)